graylog2-server/src/main/java/org/graylog2/shared/rest/documentation/generator/Generator.java

```diff
@@ -827,27 +814,28 @@ public void setDefaultValue(String defaultValue) {
 
         @JsonValue
         public Map<String, Object> jsonValue() {
-            ImmutableMap.Builder<String, Object> builder = ImmutableMap.<String, Object>builder()
-                    .put("name", name)
-                    .put("description", description)
-                    .put("required", isRequired)
-                    .put("paramType", getKind());
-
-            if (defaultValue != null) {
-                builder = builder.put("defaultValue", defaultValue);
-            }
+            final HashMap<String, Object> result = new HashMap<String, Object>() {{
```

Comment:
Do we need to create an anonymous subclass of HashMap with custom initialization block inside ? Can't we just invoke all the put methods on the "result" instance, after it gets created?

Reply:
No, both would work similar. It feels a little bit cleaner this way, not having to repeat the variable all over. What would be the benefit of using an instance / the downside of using an anonymous subclass?

Reply:
It is a personal opinion, but for me initializer blocks are much less readable. 
This code also creates additional anonymous class underneath, which does not seem to be needed.

Reply:
Interesting, for me it is the other way around. The initializer block makes it clear to me that initialization of the map is scoped to it. As long as there are no relevant technical issues around it, I would keep it that way.